src/NuGet.Core/NuGet.Protocol/Model/ServiceIndexEntry.cs

```diff
@@ -43,9 +45,17 @@ public ServiceIndexEntry(Uri serviceUri, string serviceType, SemanticVersion cli
                 throw new ArgumentNullException(nameof(clientVersion));
             }
 
+            if (packageSource != null)
+            {
+                var resourceIsHttp = serviceUri.Scheme == Uri.UriSchemeHttp && serviceUri.Scheme != Uri.UriSchemeHttps;
```

Comment:
Consider where you want to raise this event every time. 
It's probably rare, so I'd probably opt-in for the more allocation friendly approach.